File: Task_1/ros2_ws/src/my_py_pkg/my_py_pkg/subscriber_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class PySubscriber(Node):
    def __init__(self):
        super().__init__('py_subscriber_node')
        # Listens to the same "shared_topic" channel pipeline track
        self.subscription = self.create_subscription(
            String,
            'shared_topic',
            self.listener_callback,
            10)

    def listener_callback(self, msg):
        self.get_logger().info(f'Python Node Heard: "{msg.data}"')

def main(args=None):
    rclpy.init(args=args)
    node = PySubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: Task_1/ros2_ws/src/my_cpp_pkg/src/subscriber_node.cpp
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class CppSubscriber : public rclcpp::Node {
public:
    CppSubscriber() : Node("cpp_subscriber_node") {
        // Listens to the exact same "shared_topic" channel pipeline
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            "shared_topic", 10, std::bind(&CppSubscriber::topic_callback, this, std::placeholders::_1));
    }

private:
    // Triggers automatically the instant a text payload drops onto the topic
    void topic_callback(const std_msgs::msg::String::SharedPtr msg) const {
        RCLCPP_INFO(this->get_logger(), "C++ Node Heard: '%s'", msg->data.c_str());
    }
    
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char * argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CppSubscriber>());
    rclcpp::shutdown();
    return 0;
}